crates/pathfinder/src/storage/state.rs

```diff
@@ -737,7 +737,7 @@ impl StarknetEventsTable {
                         ":block_number": block_number.0,
                         ":idx": idx,
                         ":transaction_hash": &transaction.transaction_hash.0.as_be_bytes()[..],
-                        ":from_address": &transaction.contract_address.0.as_be_bytes()[..],
+                        ":from_address": &transaction.source_address().0.as_be_bytes()[..],
```

Comment:
Can DECLARE transactions even have events? If not, we should rather not do this `source_address()` thing I think.

Reply:
I don't think they have events. After all, you're not running contract code that could emit events when declaring a class.

Reply:
Waiting for a reply to confirm what you noted @kkovaacs

Reply:
Just got a confirmation that **declare** cannot produce events.